core/backend/src/IModelHost.ts

```diff
@@ -271,10 +271,13 @@ export class IModelHost {
     const platform = Platform.load();
     this.registerPlatform(platform);
 
-    const iModelClientType = iModelClient && iModelClient instanceof IModelBankClient
-      ? IModelJsNative.IModelClientType.IModelBank
-      : IModelJsNative.IModelClientType.IModelHub;
-    platform.NativeUlasClient.initialize(region, applicationType, iModelClientType);
+    let iModelClientType = IModelJsNative.IModelClientType.IModelHub;
+    let iModelClientUrl: string | undefined;
+    if (iModelClient && iModelClient instanceof IModelBankClient) {
+      iModelClientType = IModelJsNative.IModelClientType.IModelBank;
+      iModelClientUrl = iModelClient.baseUrl;
+    }
+    platform.NativeUlasClient.initialize(region, applicationType, iModelClientType, iModelClientUrl);

```

Comment:
Seems fine for now, but seems
* we should generalize iModelClientUrl to a generic UrlDiscoveryServiceUrl or something along those lines.
* The API-M team is doing away with BUDDI (the service to discover Bentley service URLs) altogether - that may eventually obviate the need for this entirely.